ods_ci/tests/Tests/400__ods_dashboard/415__ods_dashboard_projects/415__ods_dashboard_projects_permissions_mgmt.robot

```diff
@@ -76,13 +82,31 @@ Verify User Can Assign Access Permissions To User Groups
     Sleep   5s
     ${USER_C} Should Not Have Access To ${PRJ_USER_B_TITLE}
 
+Verify Project Sharing Does Not Override Dashboard Permissions
+    [Tags]                  Tier1                   Sanity                  ODS-2223
+    [Setup]                 Set RHODS Users Group To rhods-users
+    Launch Data Science Project Main Page    username=${OCP_ADMIN_USER.USERNAME}    password=${OCP_ADMIN_USER.PASSWORD}
+    ...    ocp_user_auth_type=${OCP_ADMIN_USER.AUTH_TYPE}
+    Assign Admin Permissions To User ${USER_B} in Project ${PRJ_USER_B_TITLE}
+    Assign Edit Permissions To User ${USER_C} in Project ${PRJ_USER_C_TITLE}
+    Remove User From Group    username=${USER_B}    group_name=rhods-users
+    Remove User From Group    username=${USER_B}    group_name=rhods-admins
+    Remove User From Group    username=${USER_C}    group_name=rhods-users
+    Remove User From Group    username=${USER_C}    group_name=rhods-admins
+    User ${USER_B} Should Not Be Allowed To Dashboard
+    User ${USER_C} Should Not Be Allowed To Dashboard
+    [Teardown]              Run Keywords            Set Default Access Groups Settings
+    ...                     AND                     Set User Groups For Testing
```

Comment:
wait, why running this in the test teardown? I think it's not needed. You don't see issues because there is the Suite Teardown which restores the default settings.

I believe you can remove the `[Teardown]` here and let the suite teardown to restore default user settings

Reply:
I think it is safer to run `Set User Groups For Testing` in test teardown, in case the test is **not the last one** in suite, so if another test is called afterwards, then the expected Users are already set properly.

Reply:
hmm I think you could actually avoid lines 92 and 94 because those are already executed in the Suite Setup. If you want to remove the users from the rhods-admin group as well, you can also add it to the Suite Setup (and change the Suite Teardown accordingly)

Alternatively, instead of using `Set RHODS Users Group To rhods-users` you can do `Set RHODS Users Group To ${USER_GROUP_1}` and `Set RHODS Users Group To ${USER_GROUP_1}`. This should remove lines from 91 to 95 + 98-99

Reply:
I understand now why you did it as you did, it works of course. However it think it could be cleaner as I wrote in the above comment.

If you disagree, I will approve anyways

Reply:
@bdattoma the Test case steps to automate say:

> assign "Admin" permissions to a user B who does not belong to any RHODS user group
> assign "Edit" permissions to a user C who does not belong to any RHODS user group

The existing KW  `Set RHODS Users Group To` doesn't do that exactly, as it does not specifically make the user not part of a group. It modifies `oc patch OdhDashboardConfig odh-dashboard-config`, which is a slightly different scenario. 
This is why I use KW `Remove User From Group` rhods-users and rhods-admin, which does:
`oc adm groups remove-users ${group_name} ${username}`

I think this is good enough for the scenario, we can later extend this scenario if needed.

Reply:
I know, but the suite setup already removes users from group. Besides, I'm suggesting to use `${USER_GROUP_1}`  for both admin and basic user group because the USER_B and USER_C you're using are not part of that group.